src/pybamm/models/submodels/interface/sei/sei_growth.py

```diff
@@ -217,56 +223,74 @@ def get_coupled_variables(self, variables):
         return variables
 
     def set_rhs(self, variables):
-        phase_param = self.phase_param
         domain, Domain = self.domain_Domain
 
-        if self.reaction_loc == "x-average":
-            L_sei = variables[f"X-averaged {domain} {self.reaction_name}thickness [m]"]
+        if self.reaction_loc == "interface":
+            # c_sei is an interfacial quantity [mol.m-2]
+            c_sei = variables[f"{Domain} {self.reaction_name}concentration [mol.m-2]"]
+            j_sei = variables[
+                f"{Domain} electrode {self.reaction_name}"
+                "interfacial current density [A.m-2]"
+            ]
+            a = 1
+        elif self.reaction_loc == "x-average":
+            # c_sei is a bulk quanitity [mol.m-3]
+            c_sei = variables[
+                f"X-averaged {domain} {self.reaction_name}concentration [mol.m-3]"
+            ]
             j_sei = variables[
                 f"X-averaged {domain} electrode {self.reaction_name}"
                 "interfacial current density [A.m-2]"
             ]
-
+            a = variables[
+                f"X-averaged {domain} electrode {self.phase_name}"
+                "surface area to volume ratio [m-1]"
+            ]
         else:
-            L_sei = variables[f"{Domain} {self.reaction_name}thickness [m]"]
+            c_sei = variables[f"{Domain} {self.reaction_name}concentration [mol.m-3]"]
             j_sei = variables[
                 f"{Domain} electrode {self.reaction_name}"
                 "interfacial current density [A.m-2]"
             ]
+            a = variables[
+                f"{Domain} electrode {self.phase_name}"
+                "surface area to volume ratio [m-1]"
+            ]
 
-        # The spreading term acts to spread out SEI along the cracks as they grow.
-        # For SEI on initial surface (as opposed to cracks), it is zero.
         if self.reaction == "SEI on cracks":
             if self.reaction_loc == "x-average":
-                l_cr = variables[f"X-averaged {domain} particle crack length [m]"]
-                dl_cr = variables[f"X-averaged {domain} particle cracking rate [m.s-1]"]
+                roughness = variables[f"X-averaged {domain} electrode roughness ratio"]
             else:
-                l_cr = variables[f"{Domain} particle crack length [m]"]
-                dl_cr = variables[f"{Domain} particle cracking rate [m.s-1]"]
-            spreading = dl_cr / l_cr * (self.phase_param.L_sei_crack_0 - L_sei)
-        else:
-            spreading = 0
+                roughness = variables[f"{Domain} electrode roughness ratio"]
+            a *= roughness - 1  # Replace surface area with crack area
```

Comment:
can you add a comment explaining this calculation?

Reply:
Done